datamade/how-to
# Initialize containerized Django app template

## Overview

Initialize containerized Django app template with support for Heroku, ES6, and the new `docker-entrypoint.sh` approach to running migrations.

Connects #62, #56, #22 

## Notes

With two templates in the `docker/templates` repo, I was finding it difficult to maintain the existing Cookiecutter setup using plain Docker without Docker Compose. Because of this, I went ahead and refactored the templates repo to use a Compose environment. One downside of this is that users will now _have_ to generate a Python (non-Django) repo from the `docker/templates` repo, instead of directly in the repo that needs the new configs. I think this is clearer and easier anyway, but it's a major interface change, so I'm curious to hear thoughts.

## Testing Instructions

* Follow the new Django instructions in the [Docker templates README](https://github.com/datamade/how-to/pull/71/files#diff-9434fa1914bb6162fdffedd88d83993f) to create a fresh Django app
    * All of the default cookiecutter variables should work fine
* Make sure that you can `docker-compose build` and `docker-compose up` and see your new app in the browser
    * Make sure the migrations run properly with `docker-entrypoint.sh`
    * Check the console in the browser to confirm that you see `Hello world` messages indicating ES6 compilation worked
* If you're feeling invested, you can test the new Heroku deployment docs and make sure that the template works; however this is a lot of extra steps and I'll be doing it as part of the demo anyway, so if you don't feel like it I think that's fine too
